src/shims/unix/foreign_items.rs

```diff
@@ -653,13 +652,10 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 let chunk_size = CpuAffinityMask::chunk_size(this);
 
                 if this.ptr_is_null(mask)? {
-                    let efault = this.eval_libc("EFAULT");
-                    this.set_last_error(efault)?;
-                    this.write_int(-1, dest)?;
+                    this.set_last_error_and_return(LibcError("EFAULT"), dest);
```

Comment:
I think semantically `set_last_error_and_return` is the reduction of this pattern, but please double check.

Reply:
Yes this looks good, except the trailing `?` is missing.